helix-term/src/commands.rs

```diff
@@ -638,11 +640,12 @@ fn goto_window(cx: &mut Context, align: Align) {
     let last_line = view.last_line(doc);
 
     let line = match align {
-        Align::Top => (view.offset.row + scrolloff),
-        Align::Center => (view.offset.row + (height / 2)),
-        Align::Bottom => last_line.saturating_sub(scrolloff),
+        Align::Top => (view.offset.row + scrolloff + count),
+        Align::Center => (view.offset.row + ((last_line - view.offset.row) / 2)),
+        Align::Bottom => last_line.saturating_sub((scrolloff + count).min(last_line)),
     }
-    .min(last_line.saturating_sub(scrolloff));
+    .min((view.offset.row + height - 1).saturating_sub(scrolloff))
```

Comment:
What was wrong with the previous bound? I don't think this is correct, since `view.offset + height - 1` can be past `last_line` of the document

Reply:
I already used last_line in the before computation , I think we need ensure we won't go beyond the scroll off line of the bottom. For example if the content is only half page, we subtract last_line with scroll off is pointless because it's far from scrolling.